File: Practice/AlgoBootcamp-Round2021-002-June21/CP3.1041.py
def main(n, m, t):
	print("RFP #", t, sep='')

	mp = dict()

	for _ in range(n):
		s = input()
		mp[s] = 1

	final_name = " "
	min_val = float('inf')
	count_max = -1

	for _ in range(m):
		name = input()

		money, c = map(float, input().split())
		c = int(c)
		cnt = 0

		while c > 0 :
			c -= 1
			s = input()
			cnt += mp[s]

		# print(name, money, cnt)
		if cnt > count_max or (cnt == count_max and min_val > money):
			count_max = cnt;
			min_val = money
			final_name = name

	print(final_name)
	print()

if __name__ == '__main__':
	t = 1
	while True:
		n, m = map(int, input().split())
		if n == 0 and m == 0:
			break
		else:
			main(n, m, t)
			t += 1	

File: Practice/AlgoBootcamp-Round2021-002-June21/CP3.1041.cpp
#include <bits/stdc++.h>

using namespace std;
#define vt vector
#define pb push_back
#define all(c) (c).begin(), (c).end()
#define rall(c) (c).rbegin(), (c).rend()
#define sz(x) (int)(x).size()
#define rep(i, a, b) for(int i=a; i<=b; i++)
#define down(i, a, b) for(int i=a; i>=b; i--)
#define dbg(x) cerr << "[" << #x << ": " << (x) << "] ";
#define endln cerr << "\n";
#define fi first 
#define sec second

typedef long long ll;
typedef pair<int, int> ii;
const ll INF = 1e9+1;
const int N = 1e5+1;

/* stuff you should look for
	* WRITE STUFF DOWN
	* DON'T GET STUCK ON ONE APPROACH
*/


void solve(int n, int p, int t){
	cout << "RFP #" << t << endl;
	map<string, int> mp;

	string dummy;
	getline(cin, dummy);

	rep(i, 1, n){
		string str;
		getline(cin, str);
		mp[str] = 1;
	}

	int count_ = -1;
	string name_final;
	double val = double(1e9);

	while(p--){
		string name;

		getline(cin, name);
		double money;
		int c, cnt = 0;

		cin >> money >> c;
		getline(cin, dummy);

		while(c--){
			string s;
			getline(cin, s);
			cnt += mp[s];
		}

		if(cnt > count_ || (cnt == count_ && val > money)){
			count_ = cnt;
			val = money;
			name_final = name;
		}
	}

	cout << name_final << endl;

		cout << endl;
}

int main(){
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);
	int t = 1;
	int n, p;
	while(cin >> n >> p){
		if (n == 0 && p == 0)
			break;
		else{
			solve(n, p, t);
			++t;
		}
			
	}
}
